frontend/web/components/pages/SegmentsPage.tsx

```diff
@@ -63,6 +65,12 @@ const SegmentsPage: FC<SegmentsPageType> = (props) => {
   })
   const [removeSegment] = useDeleteSegmentMutation()
   const hasHadResults = useRef(false)
+  const isLimitReached =
+    ProjectStore.getTotalSegments() >= ProjectStore.getMaxSegmentsAllowed()
```

Comment:
We are referencing data in ProjectStore however we aren't listening to any changes (e.g. project updating or changing to a different project). Are we sure this is reactive?

Reply:
The data gets updated every time the `Segmentpage` is loaded.